### PR TITLE
Ensure file objects are cleaned up on error

### PR DESCRIPTION
**Pull request by pcreech**
Plugins save then call import_content on FileContentUnit objects.
This ensures that if an error happens on import_content, we also
clean up after ourselves.

re #[1457](https://pulp.plan.io/issues/1457)
https://pulp.plan.io/issues/1457

### CODE REVIEW

**Comment by pcreech** [comment]
ok test

File: server/pulp/server/db/model/__init__.py [hunk 1]
```diff
@@ -762,6 +762,14 @@ def import_content(self, path, location=None):
         with FileStorage() as storage:
             storage.put(self, path, location)
 
+    def save_and_import_content(self, path, location=None):
```
Comment (mhrivnak) [inline server/pulp/server/db/model/__init__.py]: needs a doc block

**Comment by pcreech** [comment]
ok test

**Comment by pcreech** [comment]
ok test

File: server/pulp/server/db/model/__init__.py [hunk 2]
```diff
@@ -762,6 +763,43 @@ def import_content(self, path, location=None):
...
+        """
+        Saves this object to the database, then calls safe_import_content.
```
Comment (bmbouter) [inline server/pulp/server/db/model/__init__.py]: s/object/unit/

File: server/pulp/server/db/model/__init__.py [hunk 3]
```diff
@@ -762,6 +763,43 @@ def import_content(self, path, location=None):
...
+        :param location: The (optional) location within the unit storage path
+            where the content is to be stored.
+        :type location: str
+        """
+        self.save()
+        self.safe_import_content(path, location)
+
+    def safe_import_content(self, path, location=None):
+        """
+        If import_content raises exception, cleanup then
```
Comment (bmbouter) [inline server/pulp/server/db/model/__init__.py]: s/then continue raising/and raise/
This could probably be on 1 line, Pulp has a line width limit of 100

**Comment by bmbouter** [comment]
Two small docstring language fixes. Then go ahead and merge it.

**Pull request merged**